app/services/journal_row_updater.rb

```diff
@@ -0,0 +1,35 @@
+class JournalRowUpdater
+
+  attr_accessor :order_detail, :journal_rows
+
+  def initialize(order_detail)
+    @order_detail = order_detail
+    @journal_rows = order_detail.journal_rows
+  end
+
+  def update
+    if recreate_journal_rows?
+      recreate_journal_rows
+    else
+      update_journal_rows
+    end
+    self
+  end
+
+  def update_journal_rows
+    journal_rows.each(&:update_amount)
+  end
+
+  def recreate_journal_rows
+    if journal_rows.present?
+      order_detail.journal_rows.destroy_all
+      JournalRowBuilder.new(order_detail.journal, [order_detail]).create
```

Comment:
Should we wrap these two lines up in a transaction? If something fails in the builder, it would make me feel better if we roll back the destroy_all.